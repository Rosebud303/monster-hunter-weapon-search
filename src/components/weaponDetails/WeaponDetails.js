import React from 'react';
import './WeaponDetails.scss';
import { Link } from 'react-router-dom';

const WeaponDetails = ({type, name, minDMG, maxDMG, rarity, damageType, craftable, picture}) => {
  return (
    <div className='back'>
      <div className='details'>
        <Link className='name-and-img' to={`/image/${name}`}>
          <h2>{name}</h2>
          <img src={picture} />
          <p className='click'>| click for full image |</p>
        </Link>
        <div className='dmg'>
          <p><span className='cate1'>Min:</span> {minDMG}</p>
          <p><span className='cate2'>Max:</span> {maxDMG}</p>
          <p><span className='cate3'>DMG Type:</span> {damageType}</p>
          <p><span className='cate'>Type:</span> {type}</p>
          {
            craftable === true ? <p><span className='cate'>Craftable:</span> True</p> 
            : 
            <p><span className='cate'>Craftable:</span> False</p>
          }
          <p><span className='cate'>Rarity:</span> {rarity}</p>
        </div>
        <Link className='home' to='/'><p className='home2'>BACK</p></Link>
      </div>
    </div>
  )
}

export default WeaponDetails
