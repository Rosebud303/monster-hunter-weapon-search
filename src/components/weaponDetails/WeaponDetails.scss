.details {
  display: flex;
  justify-content: space-around;
  width: 42%;
  margin: auto;
  font-size: 20px;
  font-weight: bold;
  padding: 15px;
  background: linear-gradient( rgba(0, 0, 0, 0.7), rgba(0, 0, 0, 0.7) ), url('https://i.imgur.com/APMTCaY.jpg');
  background-repeat: no-repeat;
  opacity: .8;
  background-position: center; 
  color: white;
  box-shadow: 8px 8px 12px 1px #171414;
  border: 2px solid #0b0909;
}

h2 {
  text-align: center;
  margin-top: 0;
  font-size: 25px;
}

.cate0 {
  color: #eeff21;
  text-decoration: underline;
}

.cate1 {
  color: #00ffd4;
}

.cate2 {
  color: #00d000;
}

.cate3 {
  color: #247fff;
}

.cate {
  color: #ed8b20;
}

.back {
  position: absolute;
  width: 100%;
  font-family: monospace;
}

.name-and-img {
  background: #123345;
  padding: 5px;
  box-shadow: inset -7px 7px 20px 7px #010101d9;
  width: 12em;
}

.home2 {
  margin-top: 45px;
  font-size: 26px;
  font-weight: bolder;
}

.home {
  color: white;
  display: flex;
  justify-content: center;
  text-decoration: none;
  width: 17%;
  background-image: url('https://i.imgur.com/ven6UMn.png');
  background-repeat: no-repeat;
  opacity: 1;
  background-size: 100%;
  background-position: center; 
  &:hover {
    transition: .5s;
    background-image: url('https://i.kym-cdn.com/photos/images/original/001/315/551/b93.gif');
    background-repeat: no-repeat;
    background-size: 100%;
    background-position: center;
    color: red;
  }
}
