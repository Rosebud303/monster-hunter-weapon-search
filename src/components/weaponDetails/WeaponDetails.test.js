import React from 'react';
import { shallow } from 'enzyme';
import WeaponDetails from './WeaponDetails'


describe('WeaponDetails', () => {
  let wrapper; 
  let mockId = 1
  let mockPicture = 'picture.png'
  let mockName = 'Sword'
  let mockType = 'small-sword'
  let mockMin = 80
  let mockMax = 200
  let mockDmgType = 'fire'
  let mockRare = 1
  let mockCraft = false
  beforeEach(() => {
    wrapper = shallow (
      <WeaponDetails id={mockId}
              name={mockName}
              picture={mockPicture}
              type={mockType}
              minDMG={mockMin}
              maxDMG={mockMax}
              damageType={mockDmgType}
              rarity={mockRare}
              craftable={mockCraft}/>
    )
  });

  it('should match snap shot', () => {
    expect(wrapper).toMatchSnapshot()
  });

});