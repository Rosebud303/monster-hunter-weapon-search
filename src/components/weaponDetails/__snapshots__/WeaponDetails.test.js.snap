// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WeaponDetails should match snap shot 1`] = `
<div
  className="back"
>
  <div
    className="details"
  >
    <Link
      className="name-and-img"
      to="/image/Sword"
    >
      <h2>
        Sword
      </h2>
      <img
        src="picture.png"
      />
      <p
        className="click"
      >
        | click for full image |
      </p>
    </Link>
    <div
      className="dmg"
    >
      <p>
        <span
          className="cate1"
        >
          Min:
        </span>
         
        80
      </p>
      <p>
        <span
          className="cate2"
        >
          Max:
        </span>
         
        200
      </p>
      <p>
        <span
          className="cate3"
        >
          DMG Type:
        </span>
         
        fire
      </p>
      <p>
        <span
          className="cate"
        >
          Type:
        </span>
         
        small-sword
      </p>
      <p>
        <span
          className="cate"
        >
          Craftable:
        </span>
         False
      </p>
      <p>
        <span
          className="cate"
        >
          Rarity:
        </span>
         
        1
      </p>
    </div>
    <Link
      className="home"
      to="/weapons"
    >
      <p
        className="home2"
      >
        BACK
      </p>
    </Link>
  </div>
</div>
`;
